### PR TITLE
diff: highlight word-level changes in git diffs

### PR DESCRIPTION
**Pull request by yuja**
default style:
image

with bg color:
image

# Checklist

If applicable:
- [x] I have updated `CHANGELOG.md`
- [x] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

**Comment by joyously** [comment]
I can't see the underscores with the underlines.
How is that second image using bg color? I see only black bg.

**Comment by martinvonz** [comment]
> I can't see the underscores with the underlines.

I agree they're not easy to see. Try zooming in a bit.

> How is that second image using bg color? I see only black bg.

Only modified parts, such as the added `use` line, use background color.

**Review by martinvonz — APPROVED** [review]
Seems useful!

File: cli/src/diff_util.rs [hunk 1]
```diff
@@ -887,6 +897,60 @@ fn unified_diff_hunks<'content>(
...
+fn inline_diff_hunks<'content>(
+    left_content: &'content [u8],
+    right_content: &'content [u8],
+) -> (Vec<DiffTokenVec<'content>>, Vec<DiffTokenVec<'content>>) {
+    let mut left_lines: Vec<DiffTokenVec<'content>> = vec![];
+    let mut right_lines: Vec<DiffTokenVec<'content>> = vec![];
+    let mut left_tokens: DiffTokenVec<'content> = vec![];
+    let mut right_tokens: DiffTokenVec<'content> = vec![];
+
+    // Like Diff::default_refinement(), but doesn't try to match up contents by
```
[the remark this answers is not in the document]
Comment (yuja) [reply cli/src/diff_util.rs:911]: Does this work?
```
    // Like Diff::default_refinement(), but doesn't try to match up contents by
    // lines. We know left/right_contents have no matching lines.
```

[the remark this answers is not in the document]
Comment (martinvonz) [reply cli/src/diff_util.rs:911]: Sgtm

**Pull request merged**